pygeos/testing.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+import pygeos
+
+__all__ = ["assert_geometries_equal"]
+
+
+def _assert_nan_coords_same(x, y, decimal):
+    x_ndim = pygeos.get_coordinate_dimension(x)
+    y_ndim = pygeos.get_coordinate_dimension(y)
+    if x_ndim != y_ndim:
+        return False
+
+    x_coords = pygeos.get_coordinates(x, include_z=x_ndim == 3)
+    y_coords = pygeos.get_coordinates(y, include_z=x_ndim == 3)
+
+    # Check NaN equality
+    try:
+        assert_almost_equal(x_coords, y_coords)
```

Comment:
You probably want to pass `decimal` here?

I think `np.testing.assert_allclose` is considered as the newer, preferable method to do this instead of assert_almost_equals (which uses relative/absolute tolerance instead of number of decimals)